File: backend/models/policy.py
from typing import Optional, List
from pydantic import BaseModel, Field


class NasEntry(BaseModel):
    id: Optional[int] = None
    nasname: str = Field(..., description="NAS IP address")
    shortname: str = Field(..., max_length=32)
    type: str = "other"
    secret: str = Field(..., min_length=8)
    description: Optional[str] = None


class NasResponse(BaseModel):
    id: int
    nasname: str
    shortname: str
    type: str
    description: Optional[str] = None
    # secret deliberately omitted


class PolicyUpdate(BaseModel):
    auth_mode: Optional[str] = Field(None, pattern="^(db|ldap|hybrid)$")
    default_simultaneous_use: Optional[int] = Field(None, ge=1, le=10)
    ldap_server: Optional[str] = None
    ldap_bind_dn: Optional[str] = None
    ldap_bind_pw: Optional[str] = None  # plaintext — will be encrypted
    ldap_base_dn: Optional[str] = None
    ldap_group_dn: Optional[str] = None
    ldap_user_filter: Optional[str] = None


class PolicyResponse(BaseModel):
    auth_mode: str
    default_simultaneous_use: int
    ldap_server: Optional[str] = None
    ldap_bind_dn: Optional[str] = None
    ldap_bind_pw: str = "****"
    ldap_base_dn: Optional[str] = None
    ldap_group_dn: Optional[str] = None
    ldap_user_filter: Optional[str] = None
    radius_shared_secret: str = "****"
    nas_list: List[NasResponse] = []

File: frontend/src/api/policy.ts
import client from './client'

export interface NasEntry {
  id?: number
  nasname: string
  shortname: string
  type?: string
  secret: string
  description?: string
}

export interface PolicyResponse {
  auth_mode: string
  default_simultaneous_use: number
  ldap_server: string | null
  ldap_bind_dn: string | null
  ldap_bind_pw: string
  ldap_base_dn: string | null
  ldap_group_dn: string | null
  ldap_user_filter: string | null
  radius_shared_secret: string
  nas_list: Omit<NasEntry, 'secret'>[]
}

export async function getPolicy(): Promise<PolicyResponse> {
  const { data } = await client.get('/policy')
  return data
}

export async function updatePolicy(payload: Partial<PolicyResponse> & { ldap_bind_pw?: string }): Promise<void> {
  await client.put('/policy', payload)
}

export async function getNasList(): Promise<Omit<NasEntry, 'secret'>[]> {
  const { data } = await client.get('/nas')
  return data
}

export async function createNas(entry: NasEntry): Promise<void> {
  await client.post('/nas', entry)
}

export async function updateNas(id: number, entry: NasEntry): Promise<void> {
  await client.put(`/nas/${id}`, entry)
}

export async function deleteNas(id: number): Promise<void> {
  await client.delete(`/nas/${id}`)
}
